### PR TITLE
Patch Release 2.10.4

### PR DESCRIPTION
**Pull request by cssjoe**
https://imh-internal.atlassian.net/browse/ENG7-4790

## Summary
- Patch release 2.10.4 (version bump, changelog, upgrade notice)
- Lazy Load: match only quoted image attributes when rewriting `src` / `srcset` / `sizes`
- Cache purge: restore filterable capabilities for purge-all and purge-post (already on master; changelog / `@since` aligned for 2.10.4)

## Test plan
- [x] Confirm version constants / Stable tag are `2.10.4`
- [x] Confirm `readme.txt` / `changelog.txt` 2.10.4 entries match intended release notes
- [x] Smoke Lazy Load: images/avatars with quoted attributes still rewrite; attribute values containing quote-like fragments are left alone
- [x] Smoke editor purge: filterable purge capabilities still apply for purge-all / purge-post
- [x] Run `./vendor/bin/phpunit tests/userexperience/class-w3tc-userexperience-lazyload-mutator.php`
- [x] Run `./vendor/bin/phpunit tests/admin/class-w3tc-purge-capability-test.php`

---

> [!NOTE]
> **Medium Risk**
> Changes front-end HTML mutation and cache-purge authorization paths; behavior is covered by new tests but mis-parsing or cap checks could affect editors or image markup.
> 
> **Overview**
> **Patch release 2.10.4** bumps plugin/readme version constants and documents two fixes in changelog/readme.
> 
> **Lazy Load** rewrites `src`, `srcset`, and `sizes` only on **top-level quoted** attributes (new helpers skip matches inside other attribute values), and builds SVG placeholders via `rawurlencode` instead of hand-escaped markup. `<picture>` `<source>` tags use the same path as ``.
> 
> **Cache purge** keeps **filterable** purge-all / purge-post capabilities (`Util_Capability`, admin-bar subset, purge actions without opening full W3TC settings). **Purge current page** shows only when a post ID is detected and the user passes per-post checks; `w3tc_flush_post` authorization and flushing **fall back to** `Util_Environment::detect_post_id()` when `post_id` is missing. `@since` tags are set to 2.10.4; unit tests cover lazy-load edge cases and purge caps.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ce9a820a65d98d6d61dcaec0c14cb28b273a85c2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by cssjoe** [comment]
AWS QA matrix: passed. Ready to merge when review is complete.

File: Util_Capability.php [hunk 1]
```diff
@@ -268,7 +268,7 @@ public static function is_allowed_purge_admin_bar_item( array $item ) {
 	/**
 	 * Whether the current user may execute a purge admin-action.
 	 *
-	 * @since X.X.X
+	 * @since 2.10.4
```
Comment (jacobd91) [inline Util_Capability.php:271]: **Medium — object-level authz gap in `can_execute_purge()` for `w3tc_flush_post`**

When `post_id` is missing/`0`, the method returns role-level `can_flush_post()` only. The flush handler still resolves a target via `PgCache_Flush::flush_post()` → `Util_Environment::detect_post_id()` (including `$_GET['p']`), so a user with filterable purge-post (e.g. Editor / `edit_posts`) can purge posts they cannot `edit_post`:

`admin.php?w3tc_flush_post=1&_wpnonce=…&p=<victim>`

Nonce checks are fine — this is authorization, not CSRF. Logic is already on `master` from #1371; flagging because **2.10.4 is the first tagged release shipping it**.

**Suggested fix:** resolve the effective post ID the same way as the handler (or require `post_id > 0`) and always call `can_flush_post_id()`.

**Review by jacobd91 — COMMENTED** [review]
## Review — Patch Release 2.10.4

**Verdict: conditional go** (Lazy Load + packaging LGTM; one medium purge authz note — already on `master`, first tagged in this release).

### ENG7-4629 Lazy Load — approve
Quoted top-level `src`/`srcset`/`sizes` matching + quote-state scan correctly leaves `alt` (and other attribute values) opaque. Placeholder via `rawurlencode()` avoids delimiter breakout. Regression tests cover the avatar/`src=`-in-alt class.

### ENG7-4564 Purge caps — note (see inline)
Allowlist / `manage_options` floor for settings still look correct. Inline comment on `can_execute_purge()`: omit `post_id` + supply `p=` bypasses the `edit_post` object gate. Prefer a one-line harden before merge, or explicitly risk-accept and track a follow-up.

### AT / checks
| Check | Result |
| --- | --- |
| Version / Stable tag / changelog `2.10.4` | Pass |
| PHPUnit purge + lazyload | **26/26** (87 assertions) |
| AWS matrix (ENG7-4791/4792) | Pass (per Joe) |
| Bugbot | Pass |
| Travis PR | In progress at review time |

Nit: test plan paths say `tests/userexperience/...`; file lives under `tests/admin/class-w3tc-userexperience-lazyload-mutator.php`.

Happy to land a small authz patch on this branch if you want it in 2.10.4.

File: Util_Capability.php (hunk 1 above)
Comment (cssjoe) [reply Util_Capability.php:271]: Addressed in 4b56a264.

`can_execute_purge( 'w3tc_flush_post' )` now resolves the effective post the same way as the flush path (`post_id`, else `detect_post_id()`) and always uses `can_flush_post_id()`. The admin flush handler resolves the ID the same way before calling `w3tc_flush_post()`.

Added `test_flush_post_without_post_id_still_requires_edit_post` — PHPUnit purge suite 15/15.

**Review by jacobd91 — APPROVED** [review]
## Round 2 — Approve

Follow-ups look good. Prior medium finding on `w3tc_flush_post` (omit `post_id` + `p=` bypassing `edit_post`) is closed.

### Verified
- `4b56a264` — auth and handler both resolve via `detect_post_id()` when `post_id` is missing, then always `can_flush_post_id()`
- `ce9a820a` — admin-bar “Purge Current Page” only when `post_id > 0` (matches execution auth)
- New PHPUnit covers the `p=` path
- Re-ran purge + lazyload PHPUnit: **27/27** (92 assertions)
- Bugbot / security re-review: no new medium+ issues
- Packaging still `2.10.4`

**LGTM — approve to merge.**
